sdk/src/Core/Amazon.Runtime/Credentials/_bcl+coreclr/DefaultInstanceProfileAWSCredentials.cs

```diff
@@ -131,6 +131,11 @@ private void RenewCredentials(object unused)
 
                 lastRetrievedCredentials = newCredentials;
             }
+            catch (TaskCanceledException e)
```

Comment:
You _may_ want to use `OperationCanceledException` instead, which this derives from. I can't remember the exact scenario where you get the base class instead of the other.